### ISSUE TITLE
Refactoring Api

### ISSUE THREAD

**Issue by mehmetsefa2644** [bounswe/bounswe2017group1#172]
Hey guys, 
As you all know Android team is mainly dependent on API, and waiting for backend team to complete the API. After the API is completed, we begin to implement our tasks, 
### Please DO NOT refactor or try to optimize or add/delete/change fields of the API that has been final and corresponding Android feature's implementation is done.

If you do so, we have to stop implementing new features that we are already behind, and refactor the feature that we have already implemented. So that it will be impossible for us to complete our tasks.

### Please keep in mind that Android team is mainly dependent on API, if you change API carelessly, it would be misery for us.

**Comment by hakansirin** [comment]
Hello there, 
First of all API is incomplete and will stay incomplete at least for two weeks for now, In the meetings we might still decide on some design changes each week which you have every right to object as happened yesterday. Again then we will find a middle ground as we did yesterday. 
There is no need to create bold written huge font issues about it. 

We are as backend crew is here to help you and of course we are a one big team including all of us in the same ship. Me my friends are really careful to do these design changes as it will have minimal impact on current systems and still enhance the functionalities of the project as expected. 

However yesterday's case which were the sign up service had to have some impacts. Even though I hold the request fields same as the previous case the request form format had to be changed, so that it supports file transfers which JSON does not. I called Taha to let the rest of the android team know this change in your weekly meetings. Plus I also assigned android and frontend team as reviewers to the regarding pull request as you can see #169 here. If it were a bigger change we would decide on a meeting to let all know and got your feedbacks. But sign up is the one and only one module in the system that is apart from all others. The application would work with full functionality, even though we did't be able to sign up any new user, for the current registered users.  

It is perfectly possible to implement new features and also adopt the design changes as both other two teams do. I again want to emphasize on none of changes done are careless and we are here to help you and prevent all possible miseries. 

Thank you,
Hakan Sirin

**Comment by mehmetsefa2644** [comment]
@hakansirin  thanks for the comment, 

First of all, the issue is not about the specific (yesterday's) case, it is about the general case.

You are right that requirements and design might change, but we took a whole semester to make requirements and design mature, so now, in the implementation stage, we should try to make the necessary changes in a certain and least harmful way.
(for example 2/3 of the Android team shouldn't have realized the changes when we can not sign up to the system, for this reason I thought the changes are made in a careless way).

To sum up, you are right, we might be in a situation that we need to make changes, but before changing the necessary parts, 
- each of group member's ideas should be taken, 
- and again each of group member should be aware of the upcoming change,( the one who wants to make change can open an issue about it, issues are here for this reason too)
- and then the necessary part should be changed.

This topic concerns the completeness of our project at the final presentation, so  I found it helpful to emphasize it with opening an issue.
